### PR TITLE
Add @select helper (and @default)

### PR DESCRIPTION
**Pull request by smfoote**
This PR depends on the helpersContext introduced in #100

### CODE REVIEW

**Comment by smfoote** [comment]
Fixes #91

File: src/helpers.js [hunk 1]
```diff
@@ -6,16 +6,26 @@ let emptyFrag = function() {
...
-let truthTest = function(params, bodies, context, test) {
-  util.assert(params.val !== undefined, 'The `val` param is required for @eq, @ne, @lt, @lte, @gt, and @gte');
+let truthTest = function(context, params, bodies, helperContext, test) {
   let {key, val} = params;
+  let selectState = helperContext.get('$selectState');
+  if (!key && selectState) {
+    key = selectState.key;
+  }
+  util.assert(key !== undefined, '@eq, @ne, @lt, @lte, @gt, @gte require a `key` parameter in themselves or in a parent helper');
+  util.assert(key !== undefined, '@eq, @ne, @lt, @lte, @gt, @gte require a `val` parameter');
```
Comment (prashn64) [inline src/helpers.js]: val !== undefined

Comment (smfoote) [inline src/helpers.js]: Thanks!

File: test/acceptance/helper.js [hunk 2]
```diff
@@ -670,6 +670,92 @@ let suite = {
...
+        type: 'green',
+        category: 'color'
+      },
+      expectedDom: (() => {
+        let frag = document.createDocumentFragment();
+        frag.appendChild(document.createTextNode('A lack of '));
+        frag.appendChild(document.createTextNode('color'));
+        frag.appendChild(document.createTextNode(' here'));
+        return frag;
+      })()
```
Comment (prashn64) [inline test/acceptance/helper.js:758]: Could you add a nested select case where the inner one also has a default

**Pull request merged**